From 2b3bb7aa6851ea0212e24d00e66548764b8808fb Mon Sep 17 00:00:00 2001
From: Lokesh Vutla <lokeshvutla@ti.com>
Date: Wed, 21 Oct 2020 15:25:16 +0530
Subject: [PATCH 07/26] arm64: dts: ti: k3-am654: Introduce ABI3.x specific dts

Introduce ABI3.x specific dts that supports booting with sysfw ABI 3.x
and later.

Signed-off-by: Lokesh Vutla <lokeshvutla@ti.com>
---
 arch/arm64/boot/dts/ti/Makefile               |  1 +
 .../boot/dts/ti/k3-am65-main-abi3_x.dtsi      | 86 +++++++++++++++++++
 arch/arm64/boot/dts/ti/k3-am65-main.dtsi      |  2 +-
 .../arm64/boot/dts/ti/k3-am65-mcu-abi3_x.dtsi | 18 ++++
 .../boot/dts/ti/k3-am65-wakeup-abi3_x.dtsi    | 25 ++++++
 .../dts/ti/k3-am654-base-board-abi3_x.dts     | 16 ++++
 6 files changed, 147 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/ti/k3-am65-main-abi3_x.dtsi
 create mode 100644 arch/arm64/boot/dts/ti/k3-am65-mcu-abi3_x.dtsi
 create mode 100644 arch/arm64/boot/dts/ti/k3-am65-wakeup-abi3_x.dtsi
 create mode 100644 arch/arm64/boot/dts/ti/k3-am654-base-board-abi3_x.dts

diff --git a/arch/arm64/boot/dts/ti/Makefile b/arch/arm64/boot/dts/ti/Makefile
index ea9474ca5ff5..b952d3e730d6 100644
--- a/arch/arm64/boot/dts/ti/Makefile
+++ b/arch/arm64/boot/dts/ti/Makefile
@@ -9,6 +9,7 @@
 DTC_FLAGS += -@
 
 dtb-$(CONFIG_ARCH_K3_AM6_SOC) += k3-am654-base-board.dtb \
+	k3-am654-base-board-abi3_x.dtb \
 	k3-am654-gp.dtbo \
 	k3-am654-evm-hdmi.dtbo \
 	k3-am654-evm-oldi-lcd1evm.dtbo \
diff --git a/arch/arm64/boot/dts/ti/k3-am65-main-abi3_x.dtsi b/arch/arm64/boot/dts/ti/k3-am65-main-abi3_x.dtsi
new file mode 100644
index 000000000000..8a83a833094c
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-am65-main-abi3_x.dtsi
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for AM6 SoC Family Main Domain peripherals
+ *
+ * Copyright (C) 2020 Texas Instruments Incorporated - https://www.ti.com/
+ */
+
+&cbass_main {
+	/delete-node/ main_intr;
+	
+	intr_main_gpio: interrupt-controller0 {
+		compatible = "ti,sci-intr";
+		ti,intr-trigger-type = <1>;
+		interrupt-controller;
+		interrupt-parent = <&gic500>;
+		#interrupt-cells = <1>;
+		ti,sci = <&dmsc>;
+		ti,sci-dev-id = <100>;
+		ti,interrupt-ranges = <0 392 32>;
+	};
+};
+
+&main_navss {
+	/delete-node/ main_navss_intr;
+	/delete-node/ main_navss_inta;
+
+	main_navss_intr: interrupt-controller1 {
+		compatible = "ti,sci-intr";
+		ti,intr-trigger-type = <4>;
+		interrupt-controller;
+		interrupt-parent = <&gic500>;
+		#interrupt-cells = <1>;
+		ti,sci = <&dmsc>;
+		ti,sci-dev-id = <182>;
+		ti,interrupt-ranges = <0 64 64>,
+				      <64 448 64>;
+	};
+
+	main_udmass_inta: interrupt-controller@33d00000 {
+		compatible = "ti,sci-inta";
+		reg = <0x0 0x33d00000 0x0 0x100000>;
+		interrupt-controller;
+		interrupt-parent = <&main_navss_intr>;
+		#interrupt-cells = <3>;
+		ti,sci = <&dmsc>;
+		ti,sci-dev-id = <179>;
+		ti,interrupt-ranges = <0 0 256>;
+	};
+};
+
+&mailbox0_cluster0 {
+	interrupt-parent = <&main_navss_intr>;
+	interrupts = <436>;
+};
+
+&mailbox0_cluster1 {
+	interrupt-parent = <&main_navss_intr>;
+	interrupts = <432>;
+};
+
+&ringacc {
+	ti,sci-rm-range-gp-rings = <0x1>; /* GP ring range */
+};
+
+&main_udmap {
+	ti,sci-rm-range-tchan = <0xf>, /* TX_HCHAN */
+				<0xd>; /* TX_CHAN */
+	ti,sci-rm-range-rchan = <0xb>, /* RX_HCHAN */
+				<0xa>; /* RX_CHAN */
+	ti,sci-rm-range-rflow = <0x0>; /* GP RFLOW */
+};
+
+&main_cpts {
+	interrupts-extended = <&main_navss_intr 391>;
+	interrupt-names = "cpts";
+};
+
+&main_gpio0 {
+	interrupt-parent = <&intr_main_gpio>;
+	interrupts = <192>, <193>, <194>, <195>, <196>, <197>;
+};
+
+&main_gpio1 {
+		interrupt-parent = <&intr_main_gpio>;
+		interrupts = <200>, <201>, <202>, <203>, <204>, <205>;
+};
diff --git a/arch/arm64/boot/dts/ti/k3-am65-main.dtsi b/arch/arm64/boot/dts/ti/k3-am65-main.dtsi
index 8172f1b88038..7bb3dfbe924b 100644
--- a/arch/arm64/boot/dts/ti/k3-am65-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am65-main.dtsi
@@ -388,7 +388,7 @@
 			ti,sci-rm-range-rflow = <0x6>; /* GP RFLOW */
 		};
 
-		cpts@310d0000 {
+		main_cpts: cpts@310d0000 {
 			compatible = "ti,am65-cpts";
 			reg = <0x0 0x310d0000 0x0 0x400>;
 			reg-names = "cpts";
diff --git a/arch/arm64/boot/dts/ti/k3-am65-mcu-abi3_x.dtsi b/arch/arm64/boot/dts/ti/k3-am65-mcu-abi3_x.dtsi
new file mode 100644
index 000000000000..963ea000078d
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-am65-mcu-abi3_x.dtsi
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for AM6 SoC Family MCU Domain peripherals
+ *
+ * Copyright (C) 2020 Texas Instruments Incorporated - https://www.ti.com/
+ */
+
+&mcu_ringacc {
+	ti,sci-rm-range-gp-rings = <0x1>; /* GP ring range */
+};
+
+&mcu_udmap {
+	ti,sci-rm-range-tchan = <0xf>, /* TX_HCHAN */
+				<0xd>; /* TX_CHAN */
+	ti,sci-rm-range-rchan = <0xb>, /* RX_HCHAN */
+				<0xa>; /* RX_CHAN */
+	ti,sci-rm-range-rflow = <0x0>; /* GP RFLOW */
+};
diff --git a/arch/arm64/boot/dts/ti/k3-am65-wakeup-abi3_x.dtsi b/arch/arm64/boot/dts/ti/k3-am65-wakeup-abi3_x.dtsi
new file mode 100644
index 000000000000..04e5221d65bb
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-am65-wakeup-abi3_x.dtsi
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for AM6 SoC Family Wakeup Domain peripherals
+ *
+ * Copyright (C) 2020 Texas Instruments Incorporated - https://www.ti.com/
+ */
+
+&cbass_wakeup {
+	/delete-node/ wkup_intr;
+	intr_wkup_gpio: interrupt-controller2 {
+		compatible = "ti,sci-intr";
+		ti,intr-trigger-type = <1>;
+		interrupt-controller;
+		interrupt-parent = <&gic500>;
+		#interrupt-cells = <1>;
+		ti,sci = <&dmsc>;
+		ti,sci-dev-id = <156>;
+		ti,interrupt-ranges = <0 712 16>;
+	};
+};
+
+&wkup_gpio0 {
+	interrupt-parent = <&intr_wkup_gpio>;
+	interrupts = <60>, <61>, <62>, <63>;
+};
diff --git a/arch/arm64/boot/dts/ti/k3-am654-base-board-abi3_x.dts b/arch/arm64/boot/dts/ti/k3-am654-base-board-abi3_x.dts
new file mode 100644
index 000000000000..436d8ba877ad
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-am654-base-board-abi3_x.dts
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (C) 2016-2018 Texas Instruments Incorporated - http://www.ti.com/
+ */
+
+/dts-v1/;
+
+#include "k3-am654-base-board.dts"
+#include "k3-am65-main-abi3_x.dtsi"
+#include "k3-am65-mcu-abi3_x.dtsi"
+#include "k3-am65-wakeup-abi3_x.dtsi"
+
+/ {
+	compatible =  "ti,am654-evm", "ti,am654";
+	model = "Texas Instruments AM654 Base Board using sysfw ABI 3.x";
+};
-- 
2.31.1

